Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -22,17 +22,20 @@ namespace AzToolsFramework::ComponentModeFramework
     ComponentData::ComponentData(AZ::EntityComponentIdPair pairId)
         : m_pairId(pairId)
     {
-        AZ::ComponentApplicationBus::BroadcastResult(
-            m_entity, &AZ::ComponentApplicationBus::Events::FindEntity, AZ::EntityId(pairId.GetEntityId()));
-
-        m_component = m_entity->FindComponent(pairId.GetComponentId());
-        m_componentName = AzToolsFramework::GetFriendlyComponentName(m_component);
+        AZ::Component* component = GetComponent();
+        m_componentName = AzToolsFramework::GetFriendlyComponentName(component);
 
         AzToolsFramework::EditorRequestBus::BroadcastResult(
             m_iconPath,
             &AzToolsFramework::EditorRequestBus::Events::GetComponentEditorIcon,
-            m_component->GetUnderlyingComponentType(),
-            m_component);
+            component->GetUnderlyingComponentType(),
+            component);
+    }
+
+    AZ::Component* ComponentData::GetComponent() const
+    {
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(m_pairId.GetEntityId());
+        return entity->FindComponent(m_pairId.GetComponentId());
```

Comment:
Nit: Maybe add an assert here to validate the `entity`

Reply:
The function AzToolsFramework::GetEntityById does the assert, it's from EdtitorEntityHelpers